controller/telemetry/server_test.go

```diff
@@ -0,0 +1,193 @@
+package telemetry
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+	read "github.com/runconduit/conduit/controller/gen/controller/telemetry"
+	"golang.org/x/net/context"
+)
+
+type mockProm struct {
+	api v1.API
+	res model.Value
+}
+
+// satisfies v1.API
+func (m *mockProm) Query(ctx context.Context, query string, ts time.Time) (model.Value, error) {
+	return m.res, nil
+}
+func (m *mockProm) QueryRange(ctx context.Context, query string, r v1.Range) (model.Value, error) {
+	return m.res, nil
+}
+func (m *mockProm) LabelValues(ctx context.Context, label string) (model.LabelValues, error) {
+	return nil, nil
+}
+func (m *mockProm) Series(ctx context.Context, matches []string, startTime time.Time, endTime time.Time) ([]model.LabelSet, error) {
+	return nil, nil
+}
+
+func TestServerQueryTypes(t *testing.T) {
+
+	t.Run("Query without time range performs single metric request", func(t *testing.T) {
+		s := server{
+			prometheusAPI: &mockProm{
+				res: model.Vector{},
+			},
+		}
+		req := &read.QueryRequest{
+			Query: "fake query",
+		}
+		_, err := s.Query(context.Background(), req)
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+	})
+
+	t.Run("Query with time range performs timeseries request", func(t *testing.T) {
```

Comment:
I might be missing something, but this test (and the one in the next section) passes even if you don't include a time range. are we just testing that it accepts such an input without error?

Reply:
Really good catch !

I had previously implemented telemetry service's `Query()` endpoint to handle all querying first, then all data processes second. I felt these edge cases were worth the trade off to make the code a bit easier to read. Following your comment here, I've reconsidered, and have reorganized `Query()` to handle timeseries request/process in one block, and non-timeseries in another. This produces slightly more repeated code, but as you've pointed out, is more correct. I've modified the tests as well to validate these additional edge cases. Thanks!